### ISSUE TITLE
Titlebars are blurry/pixelated (Windows 10)

### ISSUE THREAD

**Issue by xSwagboy69x** [snowie2000/mactype#708]
Hi everyone,

MacType has always worked flawlessly system-wide for me, except for the titlebars. No matter what program, MacType renders all fonts very well except for this. I'm wondering if anyone has any ideas to fix this.

I'm using the latest builds of both Windows 10 64-bit and MacType (with the updated .dll that Snowie posted on Oct. 22), Registry mode, and the "ClearSharp" preset.

![2020_12_12 - 11 02 37 AM](https://user-images.githubusercontent.com/13977077/101988958-149e0680-3c6b-11eb-9305-1d9ea1ba2567.jpg)

![2020_12_12 - 10 57 46 AM](https://user-images.githubusercontent.com/13977077/101988962-1b2c7e00-3c6b-11eb-89cd-0c38e5107532.png)

As you can see, all other fonts are rendered well. I'm not sure if it's an .ini issue, maybe I accidentally excluded a process? Any help would be great, thanks!

**Comment by wmjordan** [comment]
This is a known issue. No workaround at this moment.

**Comment by xSwagboy69x** [comment]
> This is a known issue. No workaround at this moment.

Hi there, thanks for your reply.

I'm was 99% sure that in the past, I was able to get MacType to correctly render titlebars — so I dug through my screenshots folder and found a bunch of examples that — to my eyes — look like correctly rendered titlebars. Maybe I'm blind or missing something, but it really looks that way to me. Here are 3 examples:

![foobar2000_2019-11-08_23-26-12](https://user-images.githubusercontent.com/13977077/103113173-8db43b00-4627-11eb-932f-a0b2edf6a918.png)

![2020-02-01_01-32-09](https://user-images.githubusercontent.com/13977077/103113180-94db4900-4627-11eb-955d-dfbd35c8fcbd.jpg)

![ShareX_2019-09-12_11-30-06](https://user-images.githubusercontent.com/13977077/103113185-9a389380-4627-11eb-9b98-adf8a28c7433.png)

 The titlebars in these screenshots definitely look different than the ones in my first post — maybe it's another program or setting that's doing it but I can't imagine what that could be, MacType is the only thing I use that affects fonts. It's a mystery! Any theories?

**Comment by xSwagboy69x** [comment]
@wmjordan 

Here's another example, with an additional mystery! This is an old screenshot of my Notepad++ program

![notepad++_2020-02-01_00-16-07](https://user-images.githubusercontent.com/13977077/103113602-76764d00-4629-11eb-94fe-9fc060d9365e.jpg)

Notice the font on both the titlebars and tabs appear to be rendered by MacType! But here is a screenshot of my Notepad++ today

![2020_12_24 - 08 51 28 PM](https://user-images.githubusercontent.com/13977077/103113700-ee447780-4629-11eb-8495-910606e019d8.png)

In this example, both titlebar and tabs aren't properly rendered by MacType! I have no idea what could have changed — I suppose I'll play around with settings and hope I get lucky

**Comment by snowie2000** [comment]
@Jessamynn To render titlebars, you need to apply mactype to DWM.exe which is excluded by default.

The font you used in your notepad++ tab is a pixelated font called `Fixedsys`. MacType only works on vector fonts like Truetype or opentype. You should change your font of the tab.

**Comment by xSwagboy69x** [comment]
@snowie2000 

Thank you so much for your reply. I suppose this means that in the past, I must've removed/commented out the "UnloadDLL" section of MacType.ini ... this is plausible, because I recall adding a whole bunch of exclusions to that section after seeing that post where everyone was listing processes to exclude. I will try commenting out this section in both MacType.ini and the ClearSoft.ini that I'm using to see if that changes anything, and I'll report back.

As far as the Notepad++ tab, in its program settings (Settings > Preferences > General) I enabled the option called "Reduce" and it changed the font to something nicer. However, I wasn't able to figure out how to directly change the font. The theme stylers don't seem to respect the option.

**Comment by wmjordan** [comment]
I removed DWM.exe from both MacType.ini and my settings ini file in the MacType dir.
However, the title bars were not yet rendered by MacType.

I am running MacType in Service Mode.

**Comment by xSwagboy69x** [comment]
@wmjordan 

I did as well, and unfortunately I was not able to replicate the rendered titlebars that I use to have (as seen in the screenshots I posted above). I use Segoe UI system wide, and registry mode.

**Comment by wmjordan** [comment]
I checked the loaded DLL in DWM.exe with Process Explorer and did not found any MacType related DLL.

**Comment by xSwagboy69x** [comment]
@snowie2000 As per your suggestion, I checked to see if my mactype.ini or presets were excluding dwm.exe — they were not. Yet, when I opened mactype's process manager, it shows "Access Denied" for the DWM process. Here's a screenshot.

![2021_01_14 - 03 06 30 AM](https://user-images.githubusercontent.com/13977077/104561561-94761280-5615-11eb-9456-1d5b85b37740.jpg)

Hmm... I assume this is a permissions issue?

**Comment by xSwagboy69x** [comment]
Success! 

After a bunch of trial and error testing, I've managed to get MacType to successfully render all titlebars! 

What I did was replace the MacType64.dll file with the original .dll from the latest release (2019.1-beta6), dated **5/23/2019**. I had forgotten that I'd updated this .dll using Snowie's test file that he uploaded here from October 2020 ; upon restoring the original file, my titlebars were correctly rendered once more! I did not need to access DWM.exe, or change any font settings in Windows or in MacType.

Using Windows 10, Segoe UI font, default scaling, and **Registry Mode** in MacType. Service mode has always been inconsistent with rendering and sometimes unstable for me.

![2021_01_15 - 07 00 43 PM](https://user-images.githubusercontent.com/13977077/104789852-1bd19c00-5764-11eb-8f6d-41d28fc231ed.png)
